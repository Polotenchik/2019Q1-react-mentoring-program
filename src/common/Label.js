import React from 'react';
import classNames from 'classnames';

export const Label = (props) => {

    return (
      	<div>
        	{ props.content }
      	</div>
    );
};