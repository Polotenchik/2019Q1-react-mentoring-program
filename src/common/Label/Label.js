import React from 'react';
import classNames from 'classnames';

const Label = ({
  labelUC, labelWhite, labelBold, content,
}) => {
  const itemClass = classNames({
    label: true,
    'label-uppercase': labelUC,
    'label-white': labelWhite,
    'label-bold': labelBold,
  });

  return (
      <div className={ itemClass }>
          { content }
      </div>
  );
};

Label.defaultProps = {
  labelBold: false,
  content: 'Label',
  labelUC: false,
  labelWhite: false,
};

export default Label;
