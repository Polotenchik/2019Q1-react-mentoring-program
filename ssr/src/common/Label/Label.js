import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';

export const Label = ({ labelUC, labelWhite, labelBold, content }) => {
	const itemClass =classNames({
		'label': true,
		'label-uppercase': labelUC,
		'label-white': labelWhite,
		'label-bold': labelBold,
	});
	
    return (
      	<div className={ itemClass }>
        	{ content }
      	</div>
    );
};

Label.defaultProps = {
  labelBold: false,
  content: 'Label',
  labelUC: false,
  labelWhite: false
};