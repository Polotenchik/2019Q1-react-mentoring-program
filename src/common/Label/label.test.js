import React from 'react';
import { Label } from './Label';
import { shallow } from 'enzyme';

const mockContent = 'foo bar';

describe('<Label />', () => {
  describe('Rendering', () => {
    it('should render the label', () => {
      const wrapper = shallow(<Label content={ mockContent } />);
      expect(wrapper).toMatchSnapshot();
    });
  });
  describe('Configuration', () => {
    it('can set white color for the text', () => {
      expect(shallow(<Label type='search' />).is('.label-white')).toBe(true);
    });
    it('can transform the text to the uppercase', () => {
      expect(shallow(<Label type='search' />).is('.label-uppercase')).toBe(true);
    });
    it('can set the bold weight for the text', () => {
      expect(shallow(<Label type='info' />).is('.label-bold')).toBe(true);
    });
  });
});