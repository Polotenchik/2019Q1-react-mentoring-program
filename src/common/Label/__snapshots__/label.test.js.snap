// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Label /> Rendering should render the label 1`] = `
<div
  className={false}
>
  foo bar
</div>
`;
